File: Easy/48. Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        vector<vector<int>>ans;

        for(int i=0;i<matrix.size();i++){
            vector<int> temp;
            for(int j=matrix.size()-1;j>=0;j--)
            {
                int x = matrix[j][i];
                temp.push_back(x);
            }
            ans.push_back(temp);
            temp.clear();
        }
        matrix = ans;
    }
};

File: Medium/48. Rotate Image.java
class Solution {
    public void rotate(int[][] matrix) {
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<i;j++){
                int swap = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = swap;
            }
        }
        for(int i=0;i<matrix.length;i++)
        {
            int s=0;
            int e=matrix.length-1;
            while(s<=e)
            {
                int temp = matrix[i][s];
                matrix[i][s] = matrix[i][e];
                matrix[i][e] = temp;
                s++;
                e--;
            }
        }
    }
}
